src/polynomial.rs

```diff
@@ -221,164 +223,173 @@ pub(crate) fn poly_range_check<F: FieldElement>(start: usize, end: usize) -> Vec
     let mut p = vec![F::one()];
     let mut q = [F::zero(), F::one()];
     for i in start..end {
-        q[0] = -F::from(F::Integer::try_from(i).unwrap());
+        q[0] = -F::try_from(F::Integer::try_from(i).unwrap()).unwrap();
         p = poly_mul(&p, &q);
     }
     p
 }
 
```

Comment:
There's lots of code movement here. Can you confirm that no functional changes to these tests are intended?

Reply:
Yes, I replaced `from()` calls with `F::zero()`, `try_from()`, or `try_into()` calls, and I rewrote the random element iterator in the `poly_fft()` test to use rejection sampling upon conversion error.

Reply:
Thanks for confirming.